riscv/processor.cc

```diff
@@ -446,7 +446,7 @@ void processor_t::take_trap(trap_t& t, reg_t epc)
   // An unexpected trap - a trap when SDT is 1 - traps to M-mode
   if ((state.prv <= PRV_S && bit < max_xlen) &&
       (((vsdeleg >> bit) & 1)  || ((hsdeleg >> bit) & 1))) {
-    reg_t s = curr_virt ? state.nonvirtual_sstatus->read() : state.sstatus->read();
+    reg_t s = curr_virt ? state.sstatus->read() : state.nonvirtual_sstatus->read();
```

Comment:
`sstatus->read()` evaluates to `curr_virt ? vsstatus : sstatus`.

`nonvirtual_sstatus->read()` evaluates to `sstatus`.

So I think you just want `reg_t s = state.sstatus->read();`.

Reply:
Thanks. Updated.